nexus_keycode/protocols/small.py

```diff
@@ -286,23 +291,26 @@ def __init__(self, id_, days, secret_key):
         )
 
     def _generate_body(self, days):
-        if 1 <= days <= 90:
-            increment_id = days - 1
-        elif 91 <= days <= 180:
-            increment_id = (days - 91) / 2 + 90
-        elif 181 <= days <= 360:
-            increment_id = (days - 181) / 4 + 135
-        elif 361 <= days <= 720:
-            increment_id = (days - 361) / 8 + 180
-        elif 721 <= days <= 1184:
-            increment_id = (days - 721) / 16 + 225
-        elif days == 0:  # lock device
-            increment_id = 254
+        if isinstance(days, int):
+            if 1 <= days <= 90:
+                increment_id = days - 1
+            elif 91 <= days <= 180:
+                increment_id = (days - 91) // 2 + 90
```

Comment:
Similar comment about flooring - the protocol doesn't support those days, it might be preferable to flag the incorrect input early, and toss it out. 

(Or, we can floor, just maybe make it clear at some stage that the rounding occurred?)